.ci/validation/src/index.ts

```diff
@@ -30,38 +30,46 @@ export module SWSchemaValidator {
   export const defaultEncoding = "utf-8";
 
   export function prepareSchemas() {
-    fs.readdirSync(join(__dirname, schemaPath), {
+    const files = fs.readdirSync(path.join(__dirname, schemaPath), {
       encoding: defaultEncoding,
       recursive: false,
       withFileTypes: true,
-    }).forEach((file) => {
-      if (file.isFile()) {
-        ajv.addSchema(syncReadSchema(file.name));
-      }
     });
+
+    files
+      .filter((file) => file.isFile())
+      .forEach((file) => {
+        ajv.addSchema(syncReadSchema(file.name));
+      });
   }
 
-  function syncReadSchema(filename: string) {
-    return toJSON(join(__dirname, `${schemaPath}/${filename}`));
+  function syncReadSchema(filename: string): any {
+    return loadAsJSON(path.join(__dirname, `${schemaPath}/${filename}`));
   }
 
-  export function toJSON(filename: string) {
-    const yamlObj = yaml.load(fs.readFileSync(filename, defaultEncoding), {
+  export function loadAsJSON(filename: string): any {
+    return yamlToJSON(fs.readFileSync(filename, defaultEncoding));
+  }
+
+  export function yamlToJSON(yamlStr: string): any {
+    const yamlObj = yaml.load(yamlStr, {
       json: true,
     });
-    return JSON.parse(JSON.stringify(yamlObj, null, 2));
+    return structuredClone(yamlObj);
   }
 
-  export function validateSchema(workflow: JSON) {
+  export function validateSchema(workflow: Record<string, unknown>) {
```

Comment:
`JSON` is the type of the global `JSON` namespace that holds `JSON.parse` & `JSON.stringify`. It's type does not represent a JSON object. The correct type here is `Record<string, unknown>`